app/controllers/items_controller.rb

```diff
@@ -0,0 +1,12 @@
+class ItemsController < ApplicationController
+  def connection
```

Comment:
I'm guessing you just didn't get to refactoring this yet, but you could/should also use the Service/Facade pattern here rather than calling the API directly in the controller.